### ISSUE TITLE
Spell checker extension for SQL notebooks

### ISSUE THREAD

**Issue by jwatkin** [microsoft/azuredatastudio#15050]
**Is your feature request related to a problem? Please describe.**
I don't see a spell checker extension installation option for Azure Data Studio. This is necessary to spell check SQL notebook text cells, and to spell check comments in code cells.

**Describe the solution or feature you'd like**
Create a spell checker extension for Azure Data Studio SQL notebooks, or assist an existing spell checker author to do so. This should be able to spell check SQL notebook text cells, and to spell check comments in code cells.

**Describe alternatives you've considered**
The alternative is to copy and paste text and code into another editor that has a spell checker, but this is too cumbersome.

**Additional context**

**Comment by Charles-Gagnon** [comment]
While we don't have a spell check extension ourselves we do support many VS Code extensions and so one of those may fit your needs. 

I tried it out with https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker and was able to get spell checking for text cells when in editor mode : 

![image](https://user-images.githubusercontent.com/28519865/114221286-f9de5480-9921-11eb-86d7-476eef99a944.png)

But unfortunately it looks like it doesn't support our query editors. 

Other extensions may work though so you could try some of the other ones out there. 

To install outside extensions like this : 

1. Download the VSIX from the VS Code marketplace. You should see a "Download extension" link on the right hand side 
![image](https://user-images.githubusercontent.com/28519865/114221230-e4692a80-9921-11eb-82e7-ed0e636b7a25.png)

2. Follow the instructions [here](https://docs.microsoft.com/en-us/sql/azure-data-studio/extensions/add-extensions?view=sql-server-ver15#install-from-a-vsix) to install the extension into ADS

**Comment by jwatkin** [comment]
When I click on the Azure Data Studio Extensions icon, I'm presented with a Marketplace accordion panel that lists only 68 items. None are spell checkers. When I attempt to search for spell checkers in the Azure Data Studio "Search Extensions in Marketplace" field, I don't get any hits. It sounds like you're saying it's OK if we download Visual Studio Code extensions from outside of Azure Data Studio and then go back to Azure Data Studio to install them, is this correct? Can we break anything this way? Are all Visual Studio Code extensions safe to use with Azure Data Studio?

Thank you.

Jim

**Comment by Charles-Gagnon** [comment]
Yeah, we have a separate marketplace from VS Code so their extensions won't show up in ADS (and vice versa).

We don't guarantee that any random VS Code extension will work for ADS but many of them will without any problem (especially if they're simple ones like spell checkers). And usually the worst that can happen is the extension will fail to start so it's typically pretty safe (but this is all at your own risk regardless).

If you do find an extension that works in ADS then you could reach out to the author to see if they'd be interested in publishing it in our gallery, and point them to [these docs](https://docs.microsoft.com/sql/azure-data-studio/extensions/extension-authoring#publish-an-extension) for instructions as to how to do so.

**Comment by royzm001** [comment]
It would be great to have spellchecker built into ADS instead of downloading from Marketplace.
